src/v/archival/segment_reupload.cc

```diff
@@ -327,15 +327,15 @@ segment_collector::make_upload_candidate(
       first->index().base_timestamp(),
       io_priority_class);
 
-    if (head_seek_result.error()) {
+    if (head_seek_result.has_error()) {
         co_return upload_candidate_with_locks{upload_candidate{}, {}};
     }
 
     auto last = _segments.back();
     auto tail_seek_result = co_await storage::convert_end_offset_to_file_pos(
       _end_inclusive, last, last->index().max_timestamp(), io_priority_class);
 
-    if (tail_seek_result.error()) {
+    if (tail_seek_result.has_error()) {
```

Comment:
Ah, I see you're updating the error() calls that were in the earlier commit.  Does it make sense to squash these?